realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -243,6 +243,35 @@ static inline Obj do_create_row_with_object_id_primary_key(JNIEnv* env, jlong sh
     }
 }
 
+static inline Obj do_create_row_with_uuid_primary_key(JNIEnv* env, jlong shared_realm_ptr, jlong table_ref_ptr,
+                                                           jlong pk_column_key, jstring pk_value)
+{
+    auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
+    TableRef table = TBL_REF(table_ref_ptr);
+    ColKey col_key(pk_column_key);
+    shared_realm->verify_in_write(); // throws
+    JStringAccessor str_accessor(env, pk_value); // throws
+    if (!pk_value && !COL_NULLABLE(env, table, pk_column_key)) {
```

Comment:
We should probably mark this line as `//throws` as well. I assume `COL_NULLABLE` throws here?